### PR TITLE
[2.3][SECURITY] Add remember me cookie configuration

### PR DESCRIPTION
**Pull request by klaascuvelier**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14490 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by klaascuvelier** [comment]
Apparently this build fails on Jenkins, but it appears to be on some unrelated tests. 
All tests did succeed locally.

Any ideas how to proceed here?

File: src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php [hunk 1]
```diff
@@ -285,7 +285,18 @@ protected function cancelCookie(Request $request)
...
-        $request->attributes->set(self::COOKIE_ATTR_NAME, new Cookie($this->options['name'], null, 1, $this->options['path'], $this->options['domain']));
+        // Hard coded the default values for secure and http only, would be better if those where constants
+        $request->attributes->set(
+            self::COOKIE_ATTR_NAME,
+            new Cookie($this->options['name'],
+                null,
+                1,
+                $this->options['path'],
+                $this->options['domain'],
+                isset($this->options['secure']) ? $this->options['secure'] : false,
```
Comment (Koc) [inline src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php]: What about using `array_merge` in constructor to be sure that this options always exists?

Comment (klaascuvelier) [inline src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php]: But what is the advantage of that?

**Comment by klaascuvelier** [comment]
So all checks passed this time, not sure what the next steps are. 
cc @fabpot

**Comment by sstok** [comment]
Can you add some tests to prevent future breakage?
See #14843 for an example.

Status: needs work

**Comment by klaascuvelier** [comment]
@sstok I'll look in to it.
When I did this, should I remove the needs work label myself, so somebody takes another look at it?

**Comment by klaascuvelier** [comment]
@sstok test has been added.

File: src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php (hunk 1 above)
Comment (Koc) [inline src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php]: Big advantage for me: I see full default configuration in the constructor. So no need scroll down to line no 296 for getting default value of `secure` parameter. Also I see full list of options that available in this class.

Comment (Koc) [inline src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php]: It is common practice

https://github.com/symfony/symfony/blob/6393ec31690a3ecc73e5f1f7ea2185cda7aba203/src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php#L66
https://github.com/symfony/symfony/blob/2aeaa22e3171b5e1f10f2cc2a7ff7f6608b65f9d/src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php#L75
https://github.com/symfony/symfony/blob/aad7963ede0db55cb85290912018e4bea6b0ba7a/src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php#L82

Comment (klaascuvelier) [inline src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php]: Okay, I see your point @Koc.
Updated the PR!

**Comment by sstok** [comment]
Apply the Fabbot  patch and this should be good http://fabbot.io/report/symfony/symfony/14491/78eb0fd7b5cd9554943c610e315441715c3b629f#patchcs

**Comment by klaascuvelier** [comment]
@sstok yeah sorry about the standards, it's applied, redoing tests.

**Comment by klaascuvelier** [comment]
What will be the necessary steps to get this on the newer versions of Symfony? (2.7, 2.8)

**Comment by klaascuvelier** [comment]
Any idea why this happened? https://ci.appveyor.com/project/fabpot/symfony/build/1.0.1172/job/itg8ut8oqr82912s

**Comment by sstok** [comment]
The 2.3 branch is merged into the 2.7 and eventually 2.8 and master on a weekly basis.
Appveyor failure is unrelated.

Status: Reviewed

**Comment by klaascuvelier** [comment]
@sstok can you fill me in on what the next steps are? Do I just wait until it gets merged in, or do I have to do something else myself?

**Comment by klaascuvelier** [comment]
So apparently this one got merged instead: https://github.com/symfony/symfony/pull/14842

Kind of weird my solution was overlooked @sstok @fabpot .

**Comment by xabbuh** [comment]
@klaascuvelier I am sorry that your pull request has been overlooked. However, your solution has the advantage that it does not fail when one of the new options is omitted. Would you mind rebasing your PR on the latest `2.3` branch to modify the existing behaviour such that these options will be ignored when not passed?

**Comment by klaascuvelier** [comment]
@xabbuh it appears my tests are failing due to the merge of #16139 
I see in that ticket a remark like "oops I merged this too soon". How is this possible?

**Comment by xabbuh** [comment]
@klaascuvelier Imho you ignore those failures for the moment and just rebase your changes on the latest commit of the `2.3` branch.

**Comment by klaascuvelier** [comment]
@xabbuh, I actually did that, if I do `git fetch upstream && git merge upstream/2.3` again it tells me `Already up-to-date.` so doing another rebase seems pointless to me, no?

I might wait for this one to get merged and do another rebase? https://github.com/symfony/symfony/pull/16141

**Comment by fabpot** [comment]
#16141 has been merged now.

**Comment by xabbuh** [comment]
@klaascuvelier Well, it looks a bit strange that your PR contains the commit made by @nicolas-grekas. However, the diff looks right. Could you please also revert the changes to the test classes made in #14842 where the default values are set for the options?

**Comment by fabpot** [comment]
That's because you did a merge, not a rebase. We need a rebase to keep the history clean.

**Comment by klaascuvelier** [comment]
I did a rebase from 2.3, like described here: http://symfony.com/doc/current/contributing/code/patches.html#step-3-submit-your-patch
but with `2.3` instead of `master`

**Comment by klaascuvelier** [comment]
Can you tell me what exactly you want me to do, to fix this mess-up?

**Comment by fabpot** [comment]
This should work:

```
git fetch upstream
git rebase upstream/2.3
```

**Comment by klaascuvelier** [comment]
Okay, I did a new rebase, PR looks better now. 
The reason the other commit came in was probably because I did a `git push origin SECURE_COOKIE`, with `--force`, it failed, I git pulled and repushed. I noticed now the info on the contributing page explicitly states to use `--force`, which I did now.

**Comment by xabbuh** [comment]
This looks better. Could you also modify the test classes as described in https://github.com/symfony/symfony/pull/14491#issuecomment-145849933?

File: src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php [hunk 2]
```diff
@@ -293,7 +296,17 @@ protected function cancelCookie(Request $request)
...
+            self::COOKIE_ATTR_NAME,
+            new Cookie($this->options['name'],
+                null,
+                1,
+                $this->options['path'],
+                $this->options['domain'],
+                $this->options['secure'],
+                $this->options['httponly']
+            )
+        );
```
Comment (fabpot) [inline src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php]: should be kept on one line

**Comment by klaascuvelier** [comment]
Alright, fixed both of your remarks as well.

**Comment by xabbuh** [comment]
They should be revert in `PersistentTokenBasedRememberMeServicesTest` and `TokenBasedRememberMeServicesTest` too.

**Comment by klaascuvelier** [comment]
Anything I can do about the failing tests?

**Comment by fabpot** [comment]
Nope, they're going to be fixed by other PRs.

**Comment by fabpot** [comment]
Thank you @klaascuvelier.

**Pull request closed without merging**